File: tools/bench/answers/go_haiku/s2/10_parse_pairs.go
package main

import (
	"fmt"
	"strconv"
	"strings"
)

func main() {
	s := "a=1,b=22,c=333"
	pairs := strings.Split(s, ",")
	sum := 0
	for _, pair := range pairs {
		parts := strings.Split(pair, "=")
		val, _ := strconv.Atoi(parts[1])
		sum += val
	}
	fmt.Println(sum)
}

File: tools/bench/answers/rust_haiku/s2/10_parse_pairs.rs
fn main() {
    let input = "a=1,b=22,c=333";
    let pairs: Vec<&str> = input.split(',').collect();

    let mut sum = 0;

    for pair in pairs {
        let parts: Vec<&str> = pair.split('=').collect();
        if parts.len() == 2 {
            let val: i32 = parts[1].parse().unwrap();
            sum += val;
        }
    }

    println!("{}", sum);
}
